solver/jobs.go

```diff
@@ -374,25 +391,25 @@ func (jl *Solver) Close() {
 	jl.s.Stop()
 }
 
-func (jl *Solver) load(v, parent Vertex, j *Job) (Vertex, error) {
+func (jl *Solver) load(ctx context.Context, v, parent Vertex, j *Job) (Vertex, error) {
 	jl.mu.Lock()
 	defer jl.mu.Unlock()
 
 	cache := map[Vertex]Vertex{}
 
-	return jl.loadUnlocked(v, parent, j, cache)
+	return jl.loadUnlocked(ctx, v, parent, j, cache)
 }
 
 // called with solver lock
-func (jl *Solver) loadUnlocked(v, parent Vertex, j *Job, cache map[Vertex]Vertex) (Vertex, error) {
+func (jl *Solver) loadUnlocked(ctx context.Context, v, parent Vertex, j *Job, cache map[Vertex]Vertex) (Vertex, error) {
```

Comment:
I might be misunderstanding something, but why do we propagate the context here, but in other places uses `context.TODO`?

Reply:
For dagger in particular, this will result in our additional logrus fields (which include client IDs and such) to be propagated and added to the logs here, which is _extremely_ useful for tying together clients and the active vertexes they create/reuse. For vanilla buildkit, I'm not sure if there's any extra fields that will end up annotated, but worst case it's harmless.

The reason we don't do the same in the other logs is because those logs are in parts of the scheduler that run "async" from any actual client (they start in the scheduler loop which isn't attached to any client in particular), so context propagation is useless there.